runtime/lua/vim/lsp/util.lua

```diff
@@ -226,6 +226,165 @@ end
+    local old_char =  string.sub(old_line, old_line_length + end_col_idx + 1, old_line_length + end_col_idx + 1)
+    local new_char =  string.sub(new_line, new_line_length + end_col_idx + 1, new_line_length + end_col_idx + 1)
+    if old_char ~= new_char then
+      break
+    end
+    end_col_idx = end_col_idx - 1
+  end
+  return end_line_idx, end_col_idx
+
+end
+
+--@private
+--- Get the text of the range defined by start and end line/column
+--@param lines table list of lines
+--@param start_char integer First different character idx of range
+--@param end_char integer Last different character idx of range
+--@param start_line integer First different line idx of range
+--@param end_line integer Last different line idx of range
+--@returns string text extracted from defined region
+local function extract_text(lines, start_line, start_char, end_line, end_char)
+  if start_line == #lines + end_line + 1 then
+    if end_line == 0 then return '' end
+    local line = lines[start_line]
+    local length = #line + end_char - start_char
+    return string.sub(line, start_char, start_char + length + 1)
+  end
+  local result = string.sub(lines[start_line], start_char) .. '\n'
+  for line_idx = start_line + 1, #lines + end_line do
+    result = result .. lines[line_idx] .. '\n'
+  end
+  if end_line ~= 0 then
+    local line = lines[#lines + end_line + 1]
+    local length = #line + end_char + 1
+    result = result .. string.sub(line, 1, length)
+  end
+  return result
+end
+
+--@private
+--- Compute the length of the substituted range
+--@param lines table list of lines
+--@param start_char integer First different character idx of range
+--@param end_char integer Last different character idx of range
+--@param start_line integer First different line idx of range
+--@param end_line integer Last different line idx of range
+--@returns (int, int) end_line_idx and end_col_idx of range
+local function compute_length(lines, start_line, start_char, end_line, end_char)
+  local adj_end_line = #lines + end_line + 1
+  local adj_end_char
+  if adj_end_line > #lines then
+    adj_end_char =  end_char - 1
+  else
+    adj_end_char = #lines[adj_end_line] + end_char
+  end
+  if start_line == adj_end_line then
+    return adj_end_char - start_char + 1
+  end
+  local result = #lines[start_line] - start_char + 1
+  for line = start_line + 1, adj_end_line -1 do
+    result = result + #lines[line] + 1
+  end
+  result = result + adj_end_char + 1
+  return result
+end
+
+--- Returns the range table for the difference between old and new lines
+--@param old_lines table list of lines
+--@param new_lines table list of lines
+--@returns table start_line_idx and start_col_idx of range
+function M.compute_diff(old_lines, new_lines, start_line_idx, end_line_idx)
+  local start_line, start_char = first_difference(old_lines, new_lines, start_line_idx)
+  local end_line, end_char = last_difference(vim.list_slice(old_lines, start_line, #old_lines, end_line_idx),
```

Comment:
you're passing 4 things to `vim.list_slice` here?

Reply:
Thanks!